File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PostApprovalRequestForFlagExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PostApprovalRequestForFlagExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        config.setApiKey("YOUR_API_KEY");

        var createFlagConfigApprovalRequestRequest = new CreateFlagConfigApprovalRequestRequest();
        createFlagConfigApprovalRequestRequest.description("Requesting to update targeting");
        createFlagConfigApprovalRequestRequest.instructions(List.of ());
        createFlagConfigApprovalRequestRequest.comment("optional comment");
        createFlagConfigApprovalRequestRequest.executionDate(1706701522000L);
        createFlagConfigApprovalRequestRequest.operatingOnId("6297ed79dee7dc14e1f9a80c");
        createFlagConfigApprovalRequestRequest.notifyMemberIds(List.of (
            "1234a56b7c89d012345e678f"
        ));
        createFlagConfigApprovalRequestRequest.notifyTeamKeys(List.of (
            "example-reviewer-team"
        ));
        createFlagConfigApprovalRequestRequest.integrationConfig(null);

        try
        {
            var response = new ApprovalsApi(config).postApprovalRequestForFlag(
                null, // projectKey
                null, // featureFlagKey
                null, // environmentKey
                createFlagConfigApprovalRequestRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ApprovalsApi#postApprovalRequestForFlag");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PostApprovalRequestForFlagExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PostApprovalRequestForFlagExample
{
    fun postApprovalRequestForFlag()
    {
        ApiClient.apiKey["ApiKey"] = "YOUR_API_KEY"

        val createFlagConfigApprovalRequestRequest = CreateFlagConfigApprovalRequestRequest(
            description = "Requesting to update targeting",
            instructions = listOf (),
            comment = "optional comment",
            executionDate = 1706701522000,
            operatingOnId = "6297ed79dee7dc14e1f9a80c",
            notifyMemberIds = listOf (
                "1234a56b7c89d012345e678f",
            ),
            notifyTeamKeys = listOf (
                "example-reviewer-team",
            ),
            integrationConfig = null,
        )

        try
        {
            val response = ApprovalsApi().postApprovalRequestForFlag(
                projectKey = null,
                featureFlagKey = null,
                environmentKey = null,
                createFlagConfigApprovalRequestRequest = createFlagConfigApprovalRequestRequest,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling ApprovalsApi#postApprovalRequestForFlag")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ApprovalsApi#postApprovalRequestForFlag")
            e.printStackTrace()
        }
    }
}
